rotkehlchen/chain/evm/tokens.py

```diff
@@ -344,8 +344,19 @@ def query_tokens_for_addresses(
                 )
                 if saved_list is None:
                     continue  # Do not query if we know the address has no tokens
-                all_tokens.update(saved_list)
-                addresses_to_tokens[address] = saved_list
+
+                # get NFT tokens for address and ignores them from the query to avoid duplicates
+                cursor.execute(
+                    'SELECT identifier, blockchain FROM nfts WHERE owner_address=?',
+                    (address,),
+                )
+                excluded_addresses = {
+                    Nft(row[0]).evm_address for row in cursor
+                    if SupportedBlockchain.deserialize(row[1]) == self.evm_inquirer.blockchain
+                }
+                token_list = [x for x in saved_list if x.evm_address not in excluded_addresses]
```

Comment:
CAn't you somehow do this query in sqlite? I guess it would need substring manipulation which is gonna suck. Not sure if it's more or less performant..